[2025-07-11T15:44:22.763+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-11T15:44:22.780+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: employee_ingestion_dag.validate_data manual__2025-07-11T15:44:07.886772+00:00 [queued]>
[2025-07-11T15:44:22.785+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: employee_ingestion_dag.validate_data manual__2025-07-11T15:44:07.886772+00:00 [queued]>
[2025-07-11T15:44:22.786+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-07-11T15:44:22.794+0000] {taskinstance.py:2890} INFO - Executing <Task(_PythonDecoratedOperator): validate_data> on 2025-07-11 15:44:07.886772+00:00
[2025-07-11T15:44:22.801+0000] {warnings.py:112} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=103) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-07-11T15:44:22.803+0000] {standard_task_runner.py:72} INFO - Started process 104 to run task
[2025-07-11T15:44:22.803+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'employee_ingestion_dag', 'validate_data', 'manual__2025-07-11T15:44:07.886772+00:00', '--job-id', '50', '--raw', '--subdir', 'DAGS_FOLDER/ingestion_pipeline.py', '--cfg-path', '/tmp/tmps5dd3280']
[2025-07-11T15:44:22.804+0000] {standard_task_runner.py:105} INFO - Job 50: Subtask validate_data
[2025-07-11T15:44:22.851+0000] {task_command.py:467} INFO - Running <TaskInstance: employee_ingestion_dag.validate_data manual__2025-07-11T15:44:07.886772+00:00 [running]> on host 5a07e0ddeb4c
[2025-07-11T15:44:22.948+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='employee_ingestion_dag' AIRFLOW_CTX_TASK_ID='validate_data' AIRFLOW_CTX_EXECUTION_DATE='2025-07-11T15:44:07.886772+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-11T15:44:07.886772+00:00'
[2025-07-11T15:44:22.950+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-07-11T15:44:28.155+0000] {warnings.py:112} WARNING - /home/airflow/.local/lib/python3.12/site-packages/great_expectations/data_context/data_context/serializable_data_context.py:225: UserWarning: Warning. An existing `great_expectations.yml` was found here: /opt/airflow/gx.
    - No action was taken.
  warnings.warn(message)

[2025-07-11T15:44:28.155+0000] {warnings.py:112} WARNING - /home/airflow/.local/lib/python3.12/site-packages/great_expectations/data_context/data_context/serializable_data_context.py:233: UserWarning: Warning. An existing `config_variables.yml` was found here: /opt/airflow/gx/uncommitted.
    - No action was taken.
  warnings.warn(message)

[2025-07-11T15:44:28.181+0000] {file_data_context.py:222} INFO - FileDataContext loading fluent config
[2025-07-11T15:44:28.191+0000] {config.py:187} INFO - Loading 'datasources' ->
[{'assets': [...], 'name': 'employee_datasource', 'type': 'pandas'}]
[2025-07-11T15:44:28.192+0000] {fluent_base_model.py:276} INFO - _PandasDataAsset.dict() - missing `config_provider`, skipping config substitution
[2025-07-11T15:44:28.299+0000] {abstract_data_context.py:5516} INFO - Loaded 'employee_datasource' from fluent config
[2025-07-11T15:44:28.345+0000] {file_data_context.py:174} INFO - Saving 1 Fluent Datasources to /opt/airflow/great_expectations/great_expectations.yml
[2025-07-11T15:44:28.346+0000] {fluent_base_model.py:276} INFO - PandasDatasource.dict() - missing `config_provider`, skipping config substitution
[2025-07-11T15:44:28.367+0000] {file_data_context.py:174} INFO - Saving 1 Fluent Datasources to /opt/airflow/great_expectations/great_expectations.yml
[2025-07-11T15:44:28.368+0000] {fluent_base_model.py:276} INFO - CSVAsset.dict() - missing `config_provider`, skipping config substitution
[2025-07-11T15:44:28.368+0000] {fluent_base_model.py:276} INFO - PandasDatasource.dict() - missing `config_provider`, skipping config substitution
[2025-07-11T15:44:28.421+0000] {validator.py:1461} INFO - 	14 expectation(s) included in expectation_suite.
[2025-07-11T15:44:28.997+0000] {logging_mixin.py:190} WARNING - Calculating Metrics:   0%|          | 0/37 [00:00<?, ?it/s]
[2025-07-11T15:44:28.998+0000] {logging_mixin.py:190} WARNING - Calculating Metrics:   0%|          | 0/37 [00:00<?, ?it/s]
[2025-07-11T15:44:29.000+0000] {logging_mixin.py:190} WARNING - Calculating Metrics:   5%|5         | 2/37 [00:00<00:00, 1066.98it/s]
[2025-07-11T15:44:29.307+0000] {logging_mixin.py:190} WARNING - Calculating Metrics:   5%|5         | 2/37 [00:00<00:05,  6.48it/s]  
[2025-07-11T15:44:29.308+0000] {logging_mixin.py:190} WARNING - Calculating Metrics:   5%|5         | 2/37 [00:00<00:05,  6.48it/s]
[2025-07-11T15:44:29.309+0000] {logging_mixin.py:190} WARNING - Calculating Metrics:   8%|8         | 3/37 [00:00<00:05,  6.48it/s]
[2025-07-11T15:44:29.624+0000] {logging_mixin.py:190} WARNING - Calculating Metrics:   8%|8         | 3/37 [00:00<00:07,  4.50it/s]
[2025-07-11T15:44:29.625+0000] {logging_mixin.py:190} WARNING - Calculating Metrics:   8%|8         | 3/37 [00:00<00:07,  4.50it/s]
[2025-07-11T15:44:29.642+0000] {logging_mixin.py:190} WARNING - Calculating Metrics:  43%|####3     | 16/37 [00:00<00:04,  4.50it/s]
[2025-07-11T15:44:29.858+0000] {logging_mixin.py:190} WARNING - Calculating Metrics:  43%|####3     | 16/37 [00:00<00:00, 24.16it/s]
[2025-07-11T15:44:29.859+0000] {logging_mixin.py:190} WARNING - Calculating Metrics:  43%|####3     | 16/37 [00:00<00:00, 24.16it/s]
[2025-07-11T15:44:29.880+0000] {logging_mixin.py:190} WARNING - Calculating Metrics: 100%|##########| 37/37 [00:00<00:00, 24.16it/s]
[2025-07-11T15:44:29.966+0000] {logging_mixin.py:190} WARNING - Calculating Metrics: 100%|##########| 37/37 [00:00<00:00, 58.68it/s]
[2025-07-11T15:44:29.966+0000] {logging_mixin.py:190} WARNING - Calculating Metrics: 100%|##########| 37/37 [00:00<00:00, 58.68it/s]
[2025-07-11T15:44:29.967+0000] {logging_mixin.py:190} WARNING - Calculating Metrics: 100%|##########| 37/37 [00:00<00:00, 58.68it/s]
[2025-07-11T15:44:29.967+0000] {logging_mixin.py:190} WARNING - Calculating Metrics: 100%|##########| 37/37 [00:00<00:00, 38.17it/s]
[2025-07-11T15:44:30.458+0000] {python.py:240} INFO - Done. Returned value was: {'filepath': '/opt/airflow/data/raw_data/employee_data_with_errors_chunk_4.csv', 'total_rows': 100, 'valid_rows': 91, 'invalid_rows': 9, 'criticality': 'low', 'errors_summary': 'Age: expect_column_values_to_not_be_null failed (9)', 'error_counts': {'missing_age': 0, 'missing_monthly_income': 0, 'missing_gender': 0, 'missing_job_satisfaction': 0, 'invalid_gender': 0, 'invalid_job_role': 0, 'age_out_of_range': 0, 'income_out_of_range': 0, 'invalid_age_type': 0, 'invalid_income_type': 0, 'missing_job_role_column': 0, 'missing_Age': 9}, 'created_on': '2025-07-11T15:44:30.457937', 'bad_row_indices': [0, 10, 32, 50, 51, 60, 61, 85, 98]}
[2025-07-11T15:44:30.511+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-07-11T15:44:30.513+0000] {taskinstance.py:353} INFO - Marking task as SUCCESS. dag_id=employee_ingestion_dag, task_id=validate_data, run_id=manual__2025-07-11T15:44:07.886772+00:00, execution_date=20250711T154407, start_date=20250711T154422, end_date=20250711T154430
[2025-07-11T15:44:30.605+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-07-11T15:44:30.632+0000] {taskinstance.py:3901} INFO - 3 downstream tasks scheduled from follow-on schedule check
[2025-07-11T15:44:30.635+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
